### PR TITLE
exchanges: Remove OKGroup

### PR DESCRIPTION
**Pull request by gloriousCode**
# PR Description

With OKx using v5 and OKCoin International using V3 (with no plans for a v5 at present), there's no need for an okgroup that shares data across the same endpoints. This moves all of okgroup's functions to okcoin international and removes some of the old jank that I used to code with 4 years ago :eyes: 

This isn't a re-assessment of the whole okcoin API though, things like extra coverage hasn't been added, but I can add it if desired so we have a strong base

## How has this been tested
- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes

### CODE REVIEW

**Review by thrasher- — CHANGES_REQUESTED** [review]
Thanks for simplifying this after the OKX merge! First pass, just some minor "whilst we're here things" as I'm not expecting a full API upgrade for this PR

File: exchanges/okcoin/okcoin.go [hunk 1]
```diff
@@ -1,22 +1,749 @@
...
+}
+
+// GetAccountWithdrawalHistory retrieves all recent withdrawal records.
+func (o *OKCoin) GetAccountWithdrawalHistory(ctx context.Context, currency string) ([]WithdrawalHistoryResponse, error) {
+	var requestURL string
+	if currency != "" {
+		requestURL = getWithdrawalHistory + "/" + currency
+	} else {
+		requestURL = getWithdrawalHistory
+	}
```
Comment (shazbert) [inline exchanges/okcoin/okcoin.go:157]:
```go
requestURL := getWithdrawalHistory
if currency != "" {
		requestURL += "/" + currency
}
```

And the rest on this file.

File: exchanges/okcoin/okcoin.go [hunk 2]
```diff
@@ -1,22 +1,749 @@
...
+	if a != asset.Spot {
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, a)
+	}
+	switch a {
+	case asset.Spot:
+		endpoint = getSpotOrderBook
+		requestType = tokenSubsection
+	default:
+		return resp, errors.New("unhandled asset type")
+	}
```
Comment (shazbert) [inline exchanges/okcoin/okcoin.go:352]: Can drop this switch case as its default is handled in the if statement above,

File: exchanges/okcoin/okcoin.go [hunk 3]
```diff
@@ -1,22 +1,749 @@
...
+// GetMarginTransactionDetails Get details of the recent filled orders. Cursor pagination is used.
+// All paginated requests return the latest information (newest) as the first page sorted by newest (in chronological time) first.
+func (o *OKCoin) GetMarginTransactionDetails(ctx context.Context, request *GetSpotTransactionDetailsRequest) ([]GetSpotTransactionDetailsResponse, error) {
+	requestURL := getSpotTransactionDetails + formatParameters(request)
+	var resp []GetSpotTransactionDetailsResponse
+	return resp, o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, marginTradingSubsection, requestURL, nil, &resp, true)
+}
+
+// formatParameters Formats URL parameters, useful for optional parameters due to OKCoin signature check
+func formatParameters(request interface{}) string {
```
Comment (shazbert) [inline exchanges/okcoin/okcoin.go:590]: Suggestion: I would return an error.

File: exchanges/okcoin/okcoin_websocket.go [hunk 4]
```diff
@@ -510,16 +342,7 @@ func (o *OKGroup) wsProcessCandles(respRaw []byte) error {
...
-		var c currency.Pair
-		switch a {
-		case asset.Futures, asset.PerpetualSwap:
-			c = currency.NewPairWithDelimiter(f[0]+delimiterDash+f[1],
-				f[2],
-				currency.UnderscoreDelimiter)
-		default:
-			c = currency.NewPairWithDelimiter(f[0], f[1], delimiterDash)
-		}
+		c := currency.NewPairWithDelimiter(f[0], f[1], delimiterDash)
```
Comment (shazbert) [inline exchanges/okcoin/okcoin_websocket.go:345]: 🔥

File: exchanges/okcoin/okcoin.go [hunk 5]
```diff
@@ -1,22 +1,749 @@
...
+
+	err = o.SendPayload(ctx, request.Unset, newRequest)
+	if err != nil {
+		return err
+	}
+
+	type errCapFormat struct {
+		Error        int64  `json:"error_code,omitempty"`
+		ErrorMessage string `json:"error_message,omitempty"`
+		Result       bool   `json:"result,string,omitempty"`
```
Comment (shazbert) [inline exchanges/okcoin/okcoin.go:688]: Drop omit empty

**Review by shazbert — CHANGES_REQUESTED** [review]
Looks way better!

Some strange issue with the amount of tradeable currencies come back; 7? This endpoint seems to have all of them so maybe we can just update to this: `https://www.okcoin.com/api/v3/otc/rfq/instruments` maybe?

Nice work.

**Review by shazbert — APPROVED** [review]
tACK good stuff!

**Review by thrasher- — APPROVED** [review]
tACK, thank you for making those changes!

**Pull request merged**